statsmodels/graphics/agreement.py

```diff
@@ -0,0 +1,160 @@
+                    scatter_kwds=None,
+                    mean_line_kwds=None,
+                    limit_lines_kwds=None):
+    """
+    Tukey's Mean Difference Plot.
+
+    Tukey's Mean Difference Plot (also known as a Bland-Altman plot) is a
+    graphical method to analyze the differences between two methods of
+    measurement. The mean of the measures is plotted against their difference.
+
+    For more information see
+    https://en.wikipedia.org/wiki/Bland-Altman_plot
+
+    Parameters
+    ----------
+    m1, m2: pandas Series or array-like
+
+    sd_limit : float, default 1.96
+        The limit of agreements expressed in terms of the standard deviation of
+        the differences. If `md` is the mean of the differences, and `sd` is
+        the standard deviation of those differences, then the limits of
+        agreement that will be plotted will be
+                       md - sd_limit * sd, md + sd_limit * sd
+        The default of 1.96 will produce 95% confidence intervals for the means
+        of the differences.
+        If sd_limit = 0, no limits will be plotted, and the ylimit of the plot
+        defaults to 3 standard deviatons on either side of the mean.
+
+    ax: matplotlib.axis, optional
+        matplotlib axis object to plot on.
+
+    scatter_kwargs: keywords
+        Options to to style the scatter plot. Accepts any keywords for the
+        matplotlib Axes.scatter plotting method
+
+    mean_line_kwds: keywords
+        Options to to style the scatter plot. Accepts any keywords for the
+        matplotlib Axes.axhline plotting method
+
+    limit_lines_kwds: keywords
+        Options to to style the scatter plot. Accepts any keywords for the
+        matplotlib Axes.axhline plotting method
+
+    Returns
+    -------
+    ax: matplotlib Axis object
+
+    References
+    ----------
+    Bland JM, Altman DG (1986). "Statistical methods for assessing agreement
+    between two methods of clinical measurement"
+
+    Example
+    --------
+
+    Load relevant libraries.
+
+    >>> import statsmodels.api as sm
+    >>> import numpy as np
+    >>> import matplotlib.pyplot as plt
+
+    Making a mean difference plot.
+
+    >>> # Seed the random number generator.
+    >>> # This ensures that the results below are reproducible.
+    >>> np.random.seed(9999)
+    >>> m1 = np.random.random(20)
+    >>> m2 = np.random.random(20)
+    >>> f, ax = plt.subplots(1, figsize = (8,5))
+    >>> sm.graphics.bland_altman_plot(m1, m2, ax = ax)
+    >>> plt.show()
+
+    .. plot:: plots/graphics-mean_diff_plot.py
+    """
+    import matplotlib.pyplot as plt
+
+
+    if len(m1) != len(m2):
+        raise ValueError('m1 does not have the same length as m2.')
+    if sd_limit < 0:
+        raise ValueError('sd_limit ({}) is less than 0.'.format(sd_limit))
+
+    means = np.mean([m1, m2], axis=0)
+    diffs = m1 - m2
+    mean_diff = np.mean(diffs)
+    std_diff = np.std(diffs, axis=0)
+
+    if ax is None:
+        ax = plt.gca()
```

Comment:
we have a helper function for creating a figure with axis. 
we don't use/attach to `gca`, users can use `ax` if they want that.

Reply:
OK, let me look at the other plotting functions and modify this.